forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt

```diff
@@ -67,16 +66,12 @@ class ForageSDK : ForageSDKInterface {
             )
         )
 
-        return TokenizeCardService(
-            okHttpClient = OkHttpClientBuilder.provideOkHttpClient(
-                sessionToken,
-                merchantId,
-                idempotencyKey = UUID.randomUUID().toString(),
-                traceId = logger.getTraceIdValue()
-            ),
-            httpUrl = config.baseUrl,
-            logger = logger
-        ).tokenizeCard(
+        val tokenizeCardService = ServiceFactory(sessionToken, merchantId, logger)
+            .createTokenizeCardService(
+                idempotencyKey = UUID.randomUUID().toString()
```

Comment:
Clarifying question: Danny, by idempotent, do you mean that if you send the same PAN multiple times that it will return the same `PaymentMethod` (i.e. has the same `ref`) back?

Reply:
Not exactly. I mean that this endpoint does not care about the `IDEMPOTENCY-KEY` header at all. Our servers will discard it.

The behavior that you're describing @devinmorgan has been a hot topic recently. If a `customer_id` field is included and the same PAN is submitted twice, then it will return the same ref. Without the `customer_id` field you will always get a new ref. HOWEVER, our major partners are interested in getting the same ref back in other scenarios and we may be making a change to this in the near future.

Reply:
38e023eb77d06b5d8d287c2a826afe7d0dbb91d2